src/main/java/com/RouteController.java

```diff
@@ -114,14 +116,13 @@ public Route generateRoute(@RequestParam(value = "dest") int destID, @RequestPar
             List<Segment> shortestPath = findShortestPath(startNode, endNode);
 
             List<Node> currentRouteNodes = createRouteNodes(shortestPath, startNode);
-            Route currentRoute = new Route(currentRouteNodes, new Door(-1, "Current Location", new Coordinate(currLat, currLong, Coordinate.TYPE.GCS)), endDoor);
+            Route currentRoute = new Route(currentRouteNodes, new Door(-1, "Current Location", new Coordinate(currLat, currLong, Coordinate.TYPE.GCS)), endDoor, null);
 
             if(bestRoute == null || currentRoute.getLengthInFeet() < bestRoute.getLengthInFeet())
             {
                 bestRoute = currentRoute;
             }
         }
-
         return bestRoute;
```

Comment:
You do a null check in the previous method, but not this one. I would suggest doing it here, too.